internal/pkg/manifest/backend_svc.go

```diff
@@ -35,45 +32,24 @@ type BackendService struct {
 
 // BackendServiceConfig holds the configuration that can be overridden per environments.
 type BackendServiceConfig struct {
-	ImageConfig   imageWithPortAndHealthcheck `yaml:"image,flow"`
+	ImageConfig   ImageWithPortAndHealthcheck `yaml:"image,flow"`
 	ImageOverride `yaml:",inline"`
 	TaskConfig    `yaml:",inline"`
 	*Logging      `yaml:"logging,flow"`
 	Sidecars      map[string]*SidecarConfig `yaml:"sidecars"`
 	Network       NetworkConfig             `yaml:"network"`
 }
 
-type imageWithPortAndHealthcheck struct {
-	ImageWithPort `yaml:",inline"`
-	HealthCheck   *ContainerHealthCheck `yaml:"healthcheck"`
-}
-
-// ContainerHealthCheck holds the configuration to determine if the service container is healthy.
-// See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-healthcheck.html
-type ContainerHealthCheck struct {
-	Command     []string       `yaml:"command"`
-	Interval    *time.Duration `yaml:"interval"`
-	Retries     *int           `yaml:"retries"`
-	Timeout     *time.Duration `yaml:"timeout"`
-	StartPeriod *time.Duration `yaml:"start_period"`
-}
-
 // NewBackendService applies the props to a default backend service configuration with
 // minimal task sizes, single replica, no healthcheck, and then returns it.
 func NewBackendService(props BackendServiceProps) *BackendService {
 	svc := newDefaultBackendService()
-	var healthCheck *ContainerHealthCheck
-	if props.HealthCheck != nil {
-		// Create the healthcheck field only if the caller specified a healthcheck.
-		healthCheck = newDefaultContainerHealthCheck()
-		healthCheck.apply(props.HealthCheck)
```

Comment:
Nice! looks like you moved this logic to `(hc *ContainerHealthCheck) healthCheckOpts()` which allowed all workload types to benefit from the change 👍